Source/WebCore/platform/mac/DataDetectorHighlight.mm

```diff
@@ -145,10 +143,28 @@
 
 float DataDetectorHighlight::deviceScaleFactor() const
 {
-    if (!m_page)
+    if (!m_client)
         return 1;
 
-    return m_page->deviceScaleFactor();
+    return m_client->deviceScaleFactor();
+}
+
+bool DataDetectorHighlight::isRangeSupportingType() const
+{
+    static constexpr OptionSet rangeSupportingHighlightTypes {
+        DataDetectorHighlight::Type::TelephoneNumber,
+        DataDetectorHighlight::Type::Selection,
+        DataDetectorHighlight::Type::ImageOverlay,
+    };
+
+    return rangeSupportingHighlightTypes.contains(m_type);
+}
+
+const SimpleRange& DataDetectorHighlight::range() const
```

Comment:
de-referencing an optional without checking if it's nullptr or not seems a bit scary (even though I know the assertion should technically prevent this, but still)

Reply:
Actually, I'll keep it as-is for now because the callees that use ranges know what they're doing, and the callees that don't use ranges would never look for it (currently only needed for PDF data detection needs!)